tensorflow_probability/python/distributions/loglogistic.py

```diff
@@ -0,0 +1,151 @@
+from tensorflow_probability.python.internal import tensor_util
+
+__all__ = [
+    'LogLogistic',
+]
+
+
+class LogLogistic(transformed_distribution.TransformedDistribution):
+  """The log-logistic distribution."""
+
+  def __init__(self,
+               scale,
+               concentration,
+               validate_args=False,
+               allow_nan_stats=True,
+               name='LogLogistic'):
+    """Construct a log-logistic distribution.
+
+    The LogLogistic distribution models positive-valued random variables
+    whose logarithm is logisticly distributed with mean `log(scale)` and
+    scale `1/concentration`. It is constructed as the exponential
+    transformation of a Logistic distribution.
+
+    Args:
+      scale: Floating-point `Tensor`;
+        the scale of the log-Logistic distribution(s).
+      concentration: Floating-point `Tensor`;
+        the shape of the log- Logistic distribution(s).
+      validate_args: Python `bool`, default `False`. Whether to validate input
+        with asserts. If `validate_args` is `False`, and the inputs are
+        invalid, correct behavior is not guaranteed.
+      allow_nan_stats: Python `bool`, default `True`. If `False`, raise an
+        exception if a statistic (e.g. mean/mode/etc...) is undefined for any
+        batch member If `True`, batch members with valid parameters leading to
+        undefined statistics will return NaN for this statistic.
+      name: The name to give Ops created by the initializer.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype([scale, concentration],
+                                      dtype_hint=tf.float32)
+      self._scale = tensor_util.convert_nonref_to_tensor(
+          scale, name='scale', dtype=dtype)
+      self._concentration = tensor_util.convert_nonref_to_tensor(
+          concentration, name='concentration', dtype=dtype)
+      super(LogLogistic, self).__init__(
+          distribution=logistic.Logistic(loc=tf.math.log(self.scale),
+                                         scale=1./self.concentration),
+          bijector=exp_bijector.Exp(),
+          validate_args=validate_args,
+          parameters=parameters,
+          name=name)
+
+  @classmethod
+  def _params_event_ndims(cls):
+    return dict(scale=0, concentration=0)
+
+  @property
+  def loc(self):
+    """Distribution parameter for the pre-transformed mean."""
+    return self.distribution.loc
+
+  @property
+  def scale(self):
+    """Distribution parameter."""
+    return self._scale
+
+  @property
+  def concentration(self):
+    """Distribution parameter."""
+    return self._concentration
+
+  def _mean(self):
+    b = 1. / self.concentration
+    mean = self.scale / sinc(b)
+    return tf.where(tf.greater(self.concentration, 1.), mean, np.nan)
+
+  def _variance(self):
+    b = 1. / self.concentration
+    variance = self.scale ** 2 * (1. / sinc(2*b) - 1. / sinc(b)**2)
```

Comment:
Do you know how numerically stable this calculation is?

Reply:
I'm having a look

Reply:
There will be some numerical issues when `c -> 2`, and the only way to solve them would be to work out a numerical approximation  when `c~2` and use it with something like:

```
tf.where(c < 2.+ tol, approximation, real_value)
```
But I'm no sure if it would be a good thing to introduce a `tf.where` and a `tol`. What do you think ?